Source/Core/Common/TraversalClient.h

```diff
@@ -46,8 +47,12 @@ class TraversalClient
   State GetState() const;
   FailureReason GetFailureReason() const;
 
+  bool HasFailed() const { return GetState() == State::Failure; }
+  bool IsConnecting() const { return GetState() == State::Connecting; }
+  bool IsConnected() const { return GetState() == State::Connected; }
```

Comment:
Is there any particular reason for using `GetState()` instead of `m_state` here? Not necessarily asking you to change it, just trying to understand and learn about things I'd do differently otherwise.